apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -673,7 +669,22 @@ private Task GetNursePoolBackToHealthTask(CancellationToken cancellationToken)
 
                 // Use the cancellation token now, if the caller cancels, the task they are waiting on will be canceled,
                 // but not the nursing.
-                return newTcs.WithCancellationToken(cancellationToken);
+                return WithCancellationToken(newTcs, cancellationToken);
+
+                async Task WithCancellationToken(TaskCompletionSource<int> source, CancellationToken token)
+                {
+                    TaskCompletionSource<int> cancellable = new TaskCompletionSource<int>();
+                    token.Register(() => cancellable.TrySetCanceled());
+                    try
+                    {
+                        int dummyResult = await source.Task.ConfigureAwait(false);
```

Comment:
This doesn't propagate cancellation immediately - it'll still wait for the original task first. The final result will be the same, but with very different timing than what I'd expect. We might need a VC about this.